pylib/mozhginfo/mozhginfo/push.py

```diff
@@ -0,0 +1,62 @@
+class Push(object):
+    """This represents a Push"""
+
+    def __init__(self, date):
```

Comment:
Why are we using a dict instead of parameters?
Not sure if this was already discussed with @AutomatedTester . If you don't rebase and force push I could see the comments (force pushing deletings any feedback when looking at the diff).

We need a docstring in here and make reference to:
http://mozilla-version-control-tools.readthedocs.org/en/latest/hgmo/pushlog.html#push-objects

We should either initialize the object with one of these:
1- the pushId
2- the revision
3- the raw output from pushlog [1]
4- the full raw output from pushlog [2]

When we have initialized with either the pushId or a single revision, we will use lazy loading for the other functions if they require other data.
For #3, we will not need to load any data from pushlog *unless* it is the data that is only available through a *full* request.
For #4, we never need to load anything over the network.

For the record, we can store the raw json object as a dict and use private members to access data from it.
If the data is not available (as mentioned above) we actually talk with pushlog to get the missing data.

[1] Version 2
http://mozilla-version-control-tools.readthedocs.org/en/latest/hgmo/pushlog.html#version-2

[2]
http://mozilla-version-control-tools.readthedocs.org/en/latest/hgmo/pushlog.html#push-objects